river/main.zig

```diff
@@ -21,11 +21,13 @@ const wlr = @import("wlroots");
 const build_options = @import("build_options");
 
 const c = @import("c.zig");
-const log = @import("log.zig");
 const util = @import("util.zig");
 
 const Server = @import("Server.zig");
 
+const log = std.log.scoped(.server);
+var log_level: std.log.Level = .warn;
```

Comment:
Making this `var` doesn't affect whether or not the log level is runtime configurable. What we want to do here is set this variable to `.debug` and define a custom log function in this file which preforms the filtering by log level at runtime. The log function in river's current  `log.zig` should be suitable.

Reply:
Thanks for the feedback!
Could you elaborate on that?
I can't seem to find the reference point for `log.zig`, to see the exported functions or anything, sadly.
What do I search to find the reference?
New to Zig, and all that

Reply:
The `std.log` implementation will check for a log function defined in the root source file (`river/main.zig` in this case) and call that instead of its default implementation ([code here](https://github.com/ziglang/zig/blob/783cb980ab137d6a99c20f007d70b4191d4a63c4/lib/std/log.zig#L126)). 

We need to define such a function to make the log level configurable at runtime by passing the `-l` flag to river. Copying the log function from river's current `log.zig` to `main.zig` and making it `pub` should be sufficient.

Reply:
Took some time to figure it out, thank you very much for the explanation